include/boost/next_prior.hpp

```diff
@@ -140,6 +138,24 @@ struct prior_impl1< T, Distance, true >
     }
 };
 
+template< typename T, typename Distance, bool IsIterator = is_iterator< T >::value >
+struct prior_advance_impl :
+    public prior_minus_assign_impl< T, Distance >
+{
+};
+
+template< typename T, typename Distance >
+struct prior_advance_impl< T, Distance, true >
+{
+    static T call(T x, Distance n)
+    {
+        // Avoid negating n to sidestep possible integer overflow
+        std::reverse_iterator< T > rx(x);
```

Comment:
> I suppose, we could switch to boost::reverse_iterator, but I would prefer to avoid adding a dependency on Boost.Iterator. 

I understand the concern about the dependency on Boost.Iterator...
But if the dependency is acceptable, then I would recommend to use `boost::advance` rather than `std::advance`. With `std::advance`, `boost::next(it1, 1000)` (where `it1` is RandomAccessTraversal Readable Iterator) increments `it1` 1000 times, since `it1` is treated as InputIterator. With `boost::advance`, `boost::next(it1, 1000)` just performs `it1 + 1000`.